ext/TuringDynamicHMCExt.jl

```diff
@@ -12,10 +26,15 @@ To use it, make sure you have DynamicHMC package (version >= 2) loaded:
 using DynamicHMC
 ```
 """ 
-struct DynamicNUTS{AD,space} <: Hamiltonian{AD} end
+struct DynamicNUTS{AD,space,T<:DynamicHMC.NUTS} <: Turing.Inference.Hamiltonian{AD} 
+    sampler::T
+end
 
-DynamicNUTS(args...) = DynamicNUTS{ADBackend()}(args...)
-DynamicNUTS{AD}(space::Symbol...) where AD = DynamicNUTS{AD, space}()
+DynamicNUTS(args...) = DynamicNUTS{Turing.ADBackend()}(args...)
+DynamicNUTS{AD}(spl::DynamicHMC.NUTS, space::Tuple) where AD = DynamicNUTS{AD, space, DynamicHMC.NUTS}(spl)
```

Comment:
```suggestion
DynamicNUTS{AD}(spl::DynamicHMC.NUTS, space::Tuple) where AD = DynamicNUTS{AD, space, typeof(spl)}(spl)
```

What you are currently doing still leads to type-instability since `DynamicHMC.NUTS` is not a concrete type.